gufe/chemicalstate.py

```diff
@@ -1,6 +1,76 @@
 # This code is part of OpenFE and is licensed under the MIT license.
 # For details, see https://github.com/OpenFreeEnergy/gufe
 
+from typing import Dict
+
+import numpy as np
+from frozendict import frozendict
+
+from .custom_typing import RDKitMol
+
+
 class ChemicalState:
+    """A node of an alchemical network.
+
+    Attributes
+    ----------
+    components
+        The molecular representation of the chemical state, including
+        connectivity and coordinates. This is a frozendict with user-defined
+        labels as keys, RDKit molecules as values.
+    box_vectors
+        Numpy array indicating shape and size of unit cell for the system. May
+        be a partial definition to allow for variability on certain dimensions.
+    identifier
+        Optional identifier for the chemical state; used as part of the
+        (hashable) graph node itself when the chemical state is added to an
+        `AlchemicalNetwork`
+
+    """
+
+    def __init__(
+            self,
+            components: Dict[str, RDKitMol],
+            box_vectors: np.ndarray,
```

Comment:
So one of the discussions we had on the openfe side of things is that "box" might want to be optional, particularly accounting for the following cases:

1. Vacuum boxes
2. Boxes which need to be generated from other Boxes (i.e. mol A to mol B RBFEs essentially need to have the same number of waters / boxes, which may be more / fewer than mol B to mol C).

There's probably special considerations for membranes that I'm not thinking of right now.

Reply:
Agreed, I think having it as an optional node property makes sense, even if for vacuum boxes and boxes for RBFEs in water boxes they make more sense as an edge property. I think membrane proteins or any system where size/shape of the box is a key property is where these will have to be on the nodes.